test/chi0.jl

```diff
@@ -7,46 +7,71 @@ include("testcases.jl")
-
-        ρ0 = guess_density(basis)
-        energies, ham0 = energy_hamiltonian(basis, nothing, nothing; ρ=ρ0)
-        ρ1 = DFTK.next_density(ham0, tol=tol, eigensolver=diag_full, n_bands=n_bands).ρ
-        χ0 = compute_χ0(ham0)
-
-        # Now we make the same model, but add an artificial external potential ε * dV
-        dV = randn(eltype(basis), basis.fft_size)
-        term_builder = basis -> DFTK.TermExternal(basis, ε .* dV)
-        model = model_LDA(testcase.lattice, [spec => testcase.positions], temperature=temperature, extra_terms=[term_builder])
-        basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid, fft_size=fft_size, use_symmetry=false)
-        energies, ham = energy_hamiltonian(basis, nothing, nothing; ρ=ρ0)
-        ρ2 = DFTK.next_density(ham, tol=tol, eigensolver=diag_full, n_bands=n_bands).ρ
-        diff = (ρ2.real - ρ1.real)/ε
-
-        predicted_diff = real(reshape(χ0*vec(dV), basis.fft_size))
-        @test norm(diff - predicted_diff) < sqrt(ε)
-
-        for droptol in (0, Inf)
-            # Test the diagonal_only option
-            χ0_diag = compute_χ0(ham0; droptol=droptol)
-            diff_diag_1 = real(reshape(χ0_diag*vec(dV), basis.fft_size))
-
-            EVs = [eigen(Hermitian(Array(Hk))) for Hk in ham0.blocks]
-            Es = [EV.values[1:n_bands] for EV in EVs]
-            Vs = [EV.vectors[:, 1:n_bands] for EV in EVs]
-            occ, εF = find_occupation(basis, Es)
-            diff_diag_2 = apply_χ0(ham0, dV, Vs, εF, Es; droptol=droptol,
-                                   sternheimer_contribution=(droptol == 0))
-            @test norm(diff_diag_1 - diff_diag_2) < sqrt(ε)
+        for symmetry in (false, true)
+            for use_symmetry in (false, true)
+                model = model_LDA(testcase.lattice, [spec => testcase.positions], temperature=temperature, symmetry=(symmetry ? :force : :off))
+                basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid, fft_size=fft_size, use_symmetry=use_symmetry)
+
+                ρ0 = guess_density(basis)
+                energies, ham0 = energy_hamiltonian(basis, nothing, nothing; ρ=ρ0)
+                ρ1 = DFTK.next_density(ham0, tol=tol, eigensolver=diag_full, n_bands=n_bands).ρ
+
+                # Now we make the same model, but add an artificial external potential ε * dV
+                dV = randn(eltype(basis), basis.fft_size)
+                dV_sym = DFTK.symmetrize(from_real(basis, dV)).real
+                if symmetry
+                    dV = dV_sym
+                else
+                    @test dV_sym ≈ dV
+                end
+
+                term_builder = basis -> DFTK.TermExternal(basis, ε .* dV)
+                model = model_LDA(testcase.lattice, [spec => testcase.positions], temperature=temperature, extra_terms=[term_builder], symmetry=(symmetry ? :force : :off))
+                basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid, fft_size=fft_size, use_symmetry=use_symmetry)
```

Comment:
And again :smile:

Reply:
yeah for tests I tend to care even less than usual :-p